addon/components/search-result/component.js

```diff
@@ -0,0 +1,174 @@
+        return (this.get('result.subjects') || []).slice(0, this.get('maxSubjects'));
+    }),
+    extraSubjects: Ember.computed('result.subjects', function() {
+        return (this.get('result.subjects') || []).slice(this.get('maxSubjects'));
+    }),
+    retractionId: Ember.computed('result.lists.retractions[]', function() {
+        const retractions = this.get('result.lists.retractions');
+        if (retractions && retractions.length) {
+            return retractions[0].id;
+        }
+        return null;
+    }),
+    osfID: Ember.computed('result', function() {
+        let re = /osf.io\/(\w+)\/$/;
+        // NOTE / TODO : This will have to be removed later. Currently the only "true" preprints are solely from the OSF
+        // socarxiv and the like sometimes get picked up by as part of OSF, which is technically true. This will prevent
+        // broken links to things that aren't really preprints.
+        if (this.get('result.providers.length') === 1 && this.get('result.providers').find(provider => provider.name === 'OSF'))
+            for (let i = 0; i < this.get('result.identifiers.length'); i++)
+                if (re.test(this.get('result.identifiers')[i]))
+                    return re.exec(this.get('result.identifiers')[i])[1];
+        return false;
+    }),
+    hyperlink: Ember.computed('result', function() {
+        let re = null;
+        for (let i = 0; i < this.get('result.providers.length'); i++) {
+            //If the result has multiple providers, and one of them matches, use the first one found.
+            re = this.providerUrlRegex[this.get('result.providers')[i].name];
+            if (re) break;
+        }
+
+        re = re || this.providerUrlRegex.OSF;
+
+        const identifiers = this.get('result.identifiers').filter(ident => ident.startsWith('http://'));
+
+        for (let j = 0; j < identifiers.length; j++)
+            if (re.test(identifiers[j]))
+                return identifiers[j];
+
+        return identifiers[0];
+    }),
+    // Determines whether tags in search results should be links - preprints and registries are not using tag filter right now
+    tagsInQueryParams: Ember.computed('queryParams', function() {
+        let foundTags = false;
+        this.get('queryParams').forEach(param => {
+            if (param === 'tags') {
+                foundTags = true;
+            }
+        });
+        return foundTags;
+    }),
+    didRender() {
+        MathJax.Hub.Queue(['Typeset', MathJax.Hub, this.$()[0]]);  // jshint ignore: line
+    },
+    compare(first, second, param) {
+        if (first[param] < second[param]) {
+            return -1;
+        } else if (first[param] > second[param]) {
+            return 1;
+        } else {
+            return 0;
+        }
+    },
+    actions: {
+        addFilter(type, filter) {
+            this.sendAction('addFilter', type, filter);
+        },
+        toggleShowBody() {
+            this.set('showBody', !this.showBody);
```

Comment:
This can be simplified as `this.toggleProperty('showBody')`

For consistency, we generally use `this.get()` for all properties we fetch. (vs `this.propertyName`) Booleans are a primitive property, but if we ever changed this field to a computed, using `this.get` makes it easier to refactor.